tests/unit/test_local_project.py

```diff
@@ -358,6 +360,34 @@ def test_working_dir():
     assert project.ref == "branch"
 
 
+def test_pr_id_and_ref(tmp_path: Path):
+    """ p-s passes both ref and pr_id, we want to check out PR """
+    remote = tmp_path / "remote"
+    remote.mkdir()
+    subprocess.check_call(["git", "init", "--bare", "."], cwd=remote)
+    u = tmp_path / "upstream_git"
+    u.mkdir()
+    initiate_git_repo(u, push=True, upstream_remote=str(remote))
+    # mimic github PR
+    pr_id = "123"
+    ref = subprocess.check_output(["git", "rev-parse", "HEAD^"], cwd=u).strip().decode()
+    local_tmp_branch = "asdqwe"
+    subprocess.check_call(["git", "branch", local_tmp_branch, ref], cwd=u)
+    subprocess.check_call(
+        ["git", "push", "origin", f"{local_tmp_branch}:refs/pull/{pr_id}/head"], cwd=u
+    )
+    subprocess.check_call(["git", "branch", "-D", local_tmp_branch], cwd=u)
+
+    LocalProject(working_dir=str(u), offline=True, pr_id=pr_id, ref=ref)
+
+    assert (
+        subprocess.check_output(["git", "rev-parse", "--abbrev-ref", "HEAD"], cwd=u)
```

Comment:
```suggestion
    upstream_git = tmp_path / "upstream_git"
    upstream_git.mkdir()
    initiate_git_repo(upstream_git, push=True, upstream_remote=str(remote))
    # mimic github PR
    pr_id = "123"
    ref = subprocess.check_output(["git", "rev-parse", "HEAD^"], cwd=upstream_git).strip().decode()
    local_tmp_branch = "asdqwe"
    subprocess.check_call(["git", "branch", local_tmp_branch, ref], cwd=upstream_git)
    subprocess.check_call(
        ["git", "push", "origin", f"{local_tmp_branch}:refs/pull/{pr_id}/head"], cwd=upstream_git
    )
    subprocess.check_call(["git", "branch", "-D", local_tmp_branch], cwd=upstream_git)

    LocalProject(working_dir=str(upstream_git), offline=True, pr_id=pr_id, ref=ref)

    assert (
        subprocess.check_output(["git", "rev-parse", "--abbrev-ref", "HEAD"], cwd=upstream_git)
```

Reply:
yes

Reply:
Thanks, sorry for ruining your one-letter style..;)